### PR TITLE
use pkg_resources to allow auto-discovery of third-party strategies

### PR DESCRIPTION
**Pull request by ihaywood3**
this allows strategies to be published in external packages and made available via `pip` 
these can then be found and presented to the user during configuration

### CODE REVIEW

File: dexbot/helper.py [hunk 1]
```diff
@@ -57,3 +58,26 @@ def initialize_orders_log():
 
     if not file:
         logger.info("worker_name;ID;operation_type;base_asset;base_amount;quote_asset;quote_amount;timestamp")
+
```
Comment (mikakoi) [inline dexbot/helper.py]: Doing the try/except test inside the find_external_strategies function would make this a bit more readable, don't you think?

**Comment by MarkoPaasila** [comment]
This seems like an awesome idea. Did you have any other ideas on how to distribute strategies? I am all for this, but just wanted to ask, as I myself haven't considered this at all, and am not familiar with all different methods.

**Comment by ihaywood3** [comment]
pkg_resources is either available at load time or it isn't, so it seems
inefficient to test for it every time the function is called. I suppose I'm
pining for #ifdef

On Thu, Aug 16, 2018 at 8:13 PM, Mika Koivistoinen <notifications@github.com

**Comment by PermieBTS** [comment]
This issue has been merged..

https://github.com/Codaone/DEXBot/issues/327 has been closed

**Pull request merged**